### ISSUE TITLE
PDF multi-group table publishing fails

### ISSUE THREAD

**Issue by nikkujr** [dita-ot/dita-ot#3573]
## Expected Behavior
Each `tgroup` must be rendered correctly.

![image](https://user-images.githubusercontent.com/12742806/92199064-aba73580-eea8-11ea-8981-eefab96cd6bb.png)

## Actual Behavior
When the first `tgroup` has `cols` greater than then next one, the publishing fails. However, if you switch the two it publishes fine.

## Steps to Reproduce

Sample Topic:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
<topic id="topic_id">
    <title></title>
    <body>
        <table frame="all" rowsep="1" colsep="1" id="table_id_1">
            <title/>
            <tgroup cols="2">
                <colspec colname="c1" colnum="1" colwidth="1*"/>
                <colspec colname="c2" colnum="2" colwidth="1*"/>
                <tbody>
                    <row>
                        <entry>A</entry>
                        <entry>B</entry>
                    </row>
                </tbody>
            </tgroup>
            <tgroup cols="3">
                <colspec colname="c1" colnum="1" colwidth="1*"/>
                <colspec colname="c2" colnum="2" colwidth="1*"/>
                <colspec colname="c3" colnum="3" colwidth="1*"/>
                <tbody>
                    <row>
                        <entry>A</entry>
                        <entry>B</entry>
                        <entry>C</entry>
                    </row>
                </tbody>
            </tgroup>
        </table>
    </body>
</topic>
```
[sample.zip](https://github.com/dita-ot/dita-ot/files/5172678/sample.zip)
1. Running it in with pdf format should give you the error.

## Copy of the error message, log file or stack trace
Error: java.lang.IndexOutOfBoundsException: Index: 2, Size: 2

## Environment

* DITA-OT version: DITA-OT 3.5.2
* Operating system and version:
  _(Windows)_
* How did you run DITA-OT?
  _(`dita` command)_
* Transformation type:
  _(PDF)_

**Comment by raducoravu** [comment]
Might also be related with #3566, can you also reproduce the problem with DITA OT 3.5.1?

**Comment by nikkujr** [comment]
Yes, this might be the same issue. I will try to reproduce in DITA-OT 3.5.1

**Comment by nikkujr** [comment]
@raducoravu , This seems not to be an issue with DITA-OT 3.5.1. I was able to generate the pdf output.
![image](https://user-images.githubusercontent.com/12742806/92200463-81f00d80-eeac-11ea-8799-fabc655ed4bc.png)

**Comment by raducoravu** [comment]
@nikkujr thanks for confirming this, the problem seems to have been added in 3.5.2.

**Comment by nikkujr** [comment]
@raducoravu , Is there any workaround with this?

**Comment by raducoravu** [comment]
@nikkujr I do not. @jelovirt would this problem and the related one be good candidates for the 3.5.4 milestone?

**Comment by jelovirt** [comment]
I'll try to see if a fix is feasible for 3.5.4.
